sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableServiceLogging.java

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.tables.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.data.tables.implementation.models.RetentionPolicy;
+
+/**
+ * A model representing configurable Azure Analytics Logging settings of the Table service.
+ */
+@Fluent
+public class TableServiceLogging {
+    /*
+     * The version of Analytics to configure.
+     */
+    private String version;
+
+    /*
+     * Indicates whether all delete requests should be logged.
+     */
+    private boolean deleteLogged;
+
+    /*
+     * Indicates whether all read requests should be logged.
+     */
+    private boolean readLogged;
+
+    /*
+     * Indicates whether all write requests should be logged.
+     */
+    private boolean writeLogged;
+
+    /*
+     * The retention policy.
+     */
+    private TableServiceRetentionPolicy retentionPolicy;
+
+    /**
+     * Get the version of Analytics to configure.
+     *
+     * @return The {@code version}.
+     */
+    public String getVersion() {
+        return this.version;
+    }
+
+    /**
+     * Set the version of Analytics to configure.
+     *
+     * @param version The {@code version} to set.
+     *
+     * @return The updated {@link TableServiceLogging} object.
+     */
+    public TableServiceLogging setVersion(String version) {
```

Comment:
`setVersion` is not descriptive enough and can be confused with service version. Consider naming `setAnalyticsVersion()` .